### ISSUE TITLE
Enum type should allow for providing the VARCHAR length when using native_enum

### ISSUE THREAD

**Issue by ColdHeat** [sqlalchemy/sqlalchemy#5183]
When using https://docs.sqlalchemy.org/en/13/core/type_basics.html#sqlalchemy.types.Enum there doesn't appear to be a way to specify the length of the column that's generated if you use native_enum kwarg. 

It looks like it's calculated by the longest enum: https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/sql/sqltypes.py#L1398-L1401

This kind of runs into a similar problem with the native Enum type where it's trickier to add new entries into the Enum. 

Maybe add a `column_length` kwarg or something in that vein?

**Comment by zzzeek** [comment]
you mean...when you *dont* use the native_enum flag, that is, set native_enum=False, right?

**Comment by ColdHeat** [comment]
Oops yes you're right. I should be able to work around this with hybrid_property but this would be nice.

**Comment by zzzeek** [comment]
this would need:

1. length keyword arg

2. docstring for it

3. internal calculation of length still happens, has to check if length was given and is too short

4. unit tests for 1. send length 2. dont send length 3. send length that's too short

5. changelog file

**Comment by CaselIT** [comment]
Maybe there could also be an option to set "no limit" that may be useful for backends where the length can be left unset, like in postgres?

**Comment by zzzeek** [comment]
would an ENUM really have values that are 10000 characters long ?

**Comment by CaselIT** [comment]
No, but in pg there is no penalty for not setting the lenght of a text field. But I see your point, it's a further case to support for very little gain

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **master** branch:

**Add length parameter in `Enum`** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1815

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **rel_1_3** branch:

**Add length parameter in `Enum`** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1836
